### ISSUE TITLE
TF Lite enable Flex delegate C++ API (with CMake of Bazel)

### ISSUE THREAD

**Issue by alex19999** [tensorflow/tensorflow#57822]
**System information**
- Linux Ubuntu 18.04):
- TensorFlow installed from source:
- TensorFlow version commit 47e07ba0d68c55dba62bff5b8486291086840097:

Hi, I want to run simple conv2d layer training using C++ API of TF Lite. I’ve used examples/minimal project as reference. Calling trainer->invoke() leads to next error:

`ERROR: TensorFlow Lite Error: Select TensorFlow op(s), included in the given model is(are) not supported by this interpreter. Make sure you apply/link Flex delegate before inference. For the Android, it can be resolved by adding “org.tensorflow:tensorflow-lite-select-tf-ops” dependency…`
`ERROR: Node number 48 (FlexConv2DBackpropFilter) failed to prepare.`

I didn’t find any good tutorial on Flex delegate. Please, tell me, how to build TF Lite with Flex using CMake and how to make this interpreter use Flex’s implementation of this node. Thanks

**Comment by mohantym** [comment]
Hi @alex19999 !
Sorry for the late response.
Currently we can build Select ops flex delegates through[ Bazel](https://www.tensorflow.org/lite/guide/ops_select#cc). Not sure on Cmake .  A similar feature request has been raised here.[#55536](https://github.com/tensorflow/tensorflow/issues/55536).

I gave a try building select ops run time in cmake by changing the [build file](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/examples/minimal/BUILD) though. Could you give it a try with your model. Attached [gist](https://colab.sandbox.google.com/gist/mohantym/2510580b26892a34a8203bd2515bf092/git_57822.ipynb) for reference  (Please change the build file before running the code)
```

   deps = [
        "//tensorflow/lite:framework",
        "//tensorflow/lite/kernels:builtin_ops",
        '//tensorflow/lite/kernels:select_ops"
    ],
```

**Comment by dhruvkakadiya** [comment]
> Hi @alex19999 ! Sorry for the late response. Currently we can build Select ops flex delegates through[ Bazel](https://www.tensorflow.org/lite/guide/ops_select#cc). Not sure on building flex delegate through Cmake . A similar feature request has been raised here.[#55536](https://github.com/tensorflow/tensorflow/issues/55536).
> 
> I gave a try building select ops run time in cmake by changing the [build file](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/examples/minimal/BUILD) though. Could you give it a try with your model. Attached [gist](https://colab.sandbox.google.com/gist/mohantym/2510580b26892a34a8203bd2515bf092/git_57822.ipynb) for reference (Please change the build file before running the code)
> 
> ```
> 
>    deps = [
>         "//tensorflow/lite:framework",
>         "//tensorflow/lite/kernels:builtin_ops",
>         '//tensorflow/lite/kernels:select_ops"
>     ],
> ```

@mohantym adding `//tensorflow/lite/kernels:select_ops` entry in `deps` wouldn't end up in compilation error because there is no such target available part of `tensorflow/lite/kernels` [package](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/BUILD)?

@alex19999 I've tried just building `tensorflowlite_flex` library and linking that library while building application level binary using CMake. That's been working fine for Mac OS.

Bazel command to build `tensorflowlite_flex` library
```
bazel build -c opt --config=monolithic tensorflow/lite/delegates/flex:tensorflowlite_flex
```
Sample CMake snippet
```
add_executable(minimal minimal.cpp)
target_link_libraries(minimal tensorflowlite tensorflowlite_flex)
```

**Comment by mohantym** [comment]
@alex19999 !
That is great to hear.  Please let us know after you results with Ubuntu too.
Thank you!

**Comment by MykhailovKyrylo** [comment]
@dhruvkakadiya 
Could you please share details for this? 

> @alex19999 I've tried just building tensorflowlite_flex library and linking that library while building application level binary using CMake. That's been working fine for Mac OS.

I tried to do the same. Maybe I've missed something.

I built `tensorflowlite-flex.so` using:
```
bazel build -c opt --config=monolithic tensorflow/lite/delegates/flex:tensorflowlite_flex
```

Here's my CMake for linking:
```
add_library(tensorflowlite_flex SHARED IMPORTED)
set_property(TARGET tensorflowlite_flex PROPERTY
        IMPORTED_LOCATION "${CMAKE_CURRENT_SOURCE_DIR}/tensorflow-prebuilt/libtensorflowlite_flex.so")
        
target_link_libraries(minimal tensorflow-lite tensorflowlite_flex)
```

It fails with an error:
```
ERROR: Select TensorFlow op(s), included in the given model, is(are) not supported by this interpreter. Make sure you apply/link the Flex delegate before inference. For the Android, it can be resolved by adding "org.tensorflow:tensorflow-lite-select-tf-ops" dependency. See instructions: https://www.tensorflow.org/lite/guide/ops_select
ERROR: Node number 3 (FlexTensorArrayV3) failed to prepare.

Error at /home/kyrylomykhailov/Workspace/tensorflow/tensorflow/lite/examples/minimal/minimal.cc:61
```

Thank you!

**Comment by mohantym** [comment]
Ok @alex19999 !
Thanks for sharing your result in Ubuntu. It is still a pending feature request then.
Thank you!

**Comment by dhruvkakadiya** [comment]
Hello @MykhailovKyrylo, I've tried same process with Linux OS but yes as you observed it's not working so seems like Flex support is missing for Linux/amd64 or ARM64 like targets(with CMake) and only supported for Mac OS as @mohantym mentioned.

@mohantym can you please confirm missing feature thing for sure because then documentation in [here](https://www.tensorflow.org/lite/guide/ops_select#cc) seems misleading or are we missing some information from other documentation references?

Per table in [here](https://www.tensorflow.org/lite/guide/build_arm), can find information for `ARM` based computers that "Select TF Ops" is not supported with CMake. So that's related in this context and should we expect in future having that feature available?

**Comment by mohantym** [comment]
@dhruvkakadiya !
I have already forwarded this feature request to internal team . Will keep you posted.
Thank you!

**Comment by mohantym** [comment]
@alex19999 !
Moving this to closed status as it is duplicate to #55536 .  Please comment in the same issue.
Thank you!
